apps/emqx/src/emqx_ws_connection.erl

```diff
@@ -1013,6 +1013,10 @@ classify([Cmd = {shutdown, _Reason} | More], Packets, Cmds, Events) ->
     classify(More, Packets, [Cmd | Cmds], Events);
 classify([Cmd = close | More], Packets, Cmds, Events) ->
     classify(More, Packets, [Cmd | Cmds], Events);
+%% cowboy_websocket's close reason must be an atom to avoid crashing the sender process.
+%% The cause reasons come from parse_frame_error.
+classify([{close, #{cause := Cause}} | More], Packets, Cmds, Events) when is_atom(Cause) ->
+    classify(More, Packets, [{close, Cause} | Cmds], Events);
 classify([Cmd = {close, _Reason} | More], Packets, Cmds, Events) ->
     classify(More, Packets, [Cmd | Cmds], Events);
```

Comment:
for non-atom causes, where do they go ?

Reply:
In websocket, I only found errors like {close, #{cause => xxx}}, I haven't found other not-atom types yet, but I also haven't deleted line 1020-1021. 
So for non-atom causes it will log a  crash log as before.